content/posts/GSoC-week-2.md

```diff
@@ -0,0 +1,43 @@
+---
+
+title: "GSoC week 2 and 3"
+
+date: 2024-06-29
+
+  
+
+---
+
+![GSoC Banner](/GSoC_Banner.png)
+<img src = "/GSoC_Banner.png">
+
+This blog will be about the work I did and interesting stuff I found in these two weeks as a GSoC student.
+
+### Work so far and for the coming weeks.
+
+In these two weeks we heavily planned and implemented lot of Infrastructural changes. These changes may not effect general users of PyBaMM, but 
+developers will surely benefit a lot from them. I also had a taste of how interrelated PR's can be an issue (although not so much in my case), I'll
+explain more about it later.
+
+Let's break down the work :
+
+- Removing `run-tests.py` file : Most python projects uses a `run-tests.py`, which is used to handle command line inputs while running tests. PyBaMM already had 
+`noxfile.py`. This file is used to handle `nox` sessions, nox is a command-line tool that automates testing in multiple python environments, similar to tox. So
+we already had multiple nox sessions for different testing conditions. `noxfile` in pybamm project pointed to `run-tests.py` file and hence the work was to  completely remove the legacy code in `run-test.py` file and use nox for everything. 
+  
+
+  To do this, we had to put markers for every tests. Pytest has a feature that enables users to mark tests with specific names and that particular tests can be 
+  handled by `conftest.py` file. For example let's say I want to have tests marked with label `foo`, and to run them only when specified so. To do this we can mark 
+  the test function by adding decorator `@pytest.mark.foo` and run them with `pytest --foo`. These inputs are handled by `conftest.py` file. These markers can also be defined implicitly inside `conftest.py` file. That is an even better approach, since now we don't have to add decorators to every test function and if someone 
+  forgets to specify any marker for a test, we can still get away with it since `conftest.py` adds markets while collecting tests. You can read more about pytest markers from the official [docs](https://docs.pytest.org/en/7.1.x/example/markers.html#:~:text=with%20parametrize.-,Custom%20marker%20and%20command%20line%20option%20to%20control%20test%20runs,-%C2%B6). This is still a work in progress at time of writing this blog, you can have a [look](https://github.com/pybamm-team/PyBaMM/pull/4180) to know more about it.
```

Comment:
```suggestion
  forgets to specify any marker for a test, we can still get away with it since `conftest.py` adds markers while collecting tests. You can read more about pytest markers from the official [docs](https://docs.pytest.org/en/7.1.x/example/markers.html#:~:text=with%20parametrize.-,Custom%20marker%20and%20command%20line%20option%20to%20control%20test%20runs,-%C2%B6). This is still a work in progress at time of writing this blog, you can have a [look](https://github.com/pybamm-team/PyBaMM/pull/4180) to know more about it.
```